[CONTP-1658] update csi operator installation instructions (#36502)
DataDog/documentation: content/en/containers/kubernetes/csi_driver.md

{{% tab "Datadog Operator" %}}

<div class="alert alert-info">

> Operator-managed CSI Driver installation requires Datadog Operator <strong>v1.26.0</strong> or later. If you are using an earlier version, see <a href="#legacy-helm-based-installation">Legacy Helm-based installation</a> below.

Reviewer: Suggest removing directional language since you're already linking out to the section.

```suggestion
Operator-managed CSI Driver installation requires Datadog Operator <strong>v1.26.0</strong> or later. If you are using an earlier version, see <a href="#legacy-helm-based-installation">Legacy Helm-based installation</a>.
```

</div>

> Starting with Datadog Operator v1.26.0, the Operator can install and manage the CSI Driver on your behalf. Set `csi.enabled` to `true` in your `DatadogAgent` resource, and the Operator automatically creates a `DatadogCSIDriver` custom resource to deploy the driver:

Reviewer: ```suggestion
Starting with Datadog Operator v1.26.0, the Operator can install and manage the CSI Driver automatically. Set `csi.enabled` to `true` in your `DatadogAgent` resource, and the Operator automatically creates a `DatadogCSIDriver` custom resource to deploy the driver:
```

> Starting with Datadog Operator v1.26.0, the Operator can install and manage the CSI Driver on your behalf. Set `csi.enabled` to `true` in your `DatadogAgent` resource, and the Operator automatically creates a `DatadogCSIDriver` custom resource to deploy the driver:

Reviewer 2: @adel121 we should also surface when installing the operator to enable the ddcsi CRDs and controller:
```console
helm upgrade --install datadog-operator datadog/datadog-operator --set datadogCRDs.crds.datadogCSIDrivers=true --set datadogCSIDriver.enabled=true
```
either through the operator values.yaml or directly with `--set` flags

Author: Waiting for the 1.27 release to be done.

```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  global:
    credentials:
      apiSecret:
        secretName: datadog-secret
        keyName: api-key
    csi:
      enabled: true
```

> By default, `csi.autoManage` is `true`, which means the Operator owns the lifecycle of the `DatadogCSIDriver` resource. You can optionally customize scheduling through the `DatadogAgent` spec:

Reviewer: ```suggestion
By default, `csi.autoManage` is `true`, which means the Operator owns the life cycle of the `DatadogCSIDriver` resource. To customize scheduling, add the following `DatadogAgent` spec:
```

               cpu: 100m
               memory: 128Mi
             limits:
               cpu: 200m
               memory: 256Mi
         csi-node-driver-registrar:
           resources:
             requests:
               cpu: 50m
               memory: 64Mi
             limits:
               cpu: 100m
               memory: 128Mi
   ```

   See the full [DatadogCSIDriver CRD reference](https://github.com/DataDog/datadog-operator/blob/main/bundle/manifests/datadoghq.com_datadogcsidrivers.yaml) for all available fields.

{{% /collapse-content %}}

<div class="alert alert-warning">
<strong>Migrating from Helm-based CSI Driver installation</strong>
<p>If you previously installed the CSI Driver with the standalone Helm chart, Datadog recommends migrating to Operator-managed installation. Choose one of the following approaches:</p>
<ul>
<li><strong>Let the Operator manage the CSI Driver</strong>: Uninstall the Helm chart (<code>helm uninstall datadog-csi-driver</code>) and keep the default values for <code>csi.enabled</code> and <code>csi.autoManage</code>. The Operator automatically creates a new <code>DatadogCSIDriver</code> resource and deploys the driver.</li>

> <li><strong>Keep managing the CSI Driver with Helm</strong>: No action is required. The Operator detects the existing <code>k8s.csi.datadoghq.com</code> CSIDriver and defers to it, regardless of the <code>csi.autoManage</code> value. This backward-compatibility guarantee means your Helm-managed driver continues to run without interference. You can optionally set <code>csi.autoManage: false</code> to make this intent explicit in your configuration.</li>

Reviewer: ```suggestion
<li><strong>Keep managing the CSI Driver with Helm</strong>: No action is required. The Operator detects the existing <code>k8s.csi.datadoghq.com</code> CSIDriver and defers to it, regardless of the <code>csi.autoManage</code> value. The Operator does not interfere with your existing Helm-managed driver. To make this intent explicit, set <code>csi.autoManage: false</code>.</li>
```
